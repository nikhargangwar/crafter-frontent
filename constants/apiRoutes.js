const userRoute = 'api/user/';

export default { userRoute };