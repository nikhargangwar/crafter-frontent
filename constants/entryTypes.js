const entryTypes = {
  userConfig :{
    FRONTEND:{
      port:'integer',
      numberOfReplicas : 'integer',
      name:'string',
    },
    BACKEND:{
      port:'integer',
      numberOfReplicas:'integer',
      name:'string',
    },
    DATABASE:{
      port:'integer',
      numberOfReplicas:'integer',
      dbUser:'string',
      dbPassword:'string',
      name:'string',
      schemaName :'string',
    }},
  dockerConfig:{
    username:'string',
    email:'e-mail',
    repositoryImageAddress : 'string',
    token:'string'},
};

export default entryTypes;
