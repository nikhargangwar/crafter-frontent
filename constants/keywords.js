const arrayOfWordsForLandingPage = [ ' containerized applications on Kubernetes with our intuitive K8s Designer tool.', ' Frontend - React.', ' Backend - Node.js.', ' Database - Postgres.' ];

export default arrayOfWordsForLandingPage;