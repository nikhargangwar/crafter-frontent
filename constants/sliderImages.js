import loignSliderImg1 from '../assets/undraw_code_thinking_re_gka2.svg';
import loginSliderImg2 from '../assets/undraw_dev_focus_re_6iwt.svg';
import loginSliderImg3 from '../assets/undraw_specs_re_546x.svg';
import loginSliderImg4 from '../assets/undraw_coding_re_iv62.svg';
import signupSliderImg1 from '../assets/undraw_programmer_re_owql.svg';
import signupSliderImg2 from '../assets/undraw_programming_re_kg9v.svg';
import signupSliderImg3 from '../assets/undraw_dev_productivity_re_fylf.svg';
import signupSliderImg4 from '../assets/undraw_in_the_office_re_jtgc.svg';


const loginImages = [
  {
    original: loignSliderImg1
  },
  {
    original: loginSliderImg2
  },
  {
    original: loginSliderImg3
  },
  {
    original: loginSliderImg4
  }
];

const signupImages = [
  {
    original: signupSliderImg1
  },
  {
    original: signupSliderImg2
  },
  {
    original: signupSliderImg3
  },
  {
    original: signupSliderImg4
  }
];

export default { loginImages, signupImages };