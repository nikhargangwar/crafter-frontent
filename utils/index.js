import CustomNode from './CustomNode';
import FormValidation from './FormValidation';
import makeRequest from './makeRequest';
import {camelize, getConfiguration, getSpaceSeparated, updateConfiguration} from './ServiceHandler';
export { CustomNode, FormValidation, makeRequest, camelize, getConfiguration, getSpaceSeparated, updateConfiguration };