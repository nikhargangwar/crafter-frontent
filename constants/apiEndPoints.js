import apiRoutes from './apiRoutes';

const GET_ALL_BLUEPRINTS = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/blueprints',
  method: 'GET',
});

const GET_BLUEPRINT_METADATA = (blueprintId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}`,
  method: 'GET',
});

const GET_IMAGES =(name)=>({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/minio/images/${name}`,
  method: 'GET',
});

const GET_SERVICES_BY_BLUEPRINT_ID = (blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services`,
  method: 'GET',
});

const CREATE_BLUEPRINT = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/blueprints',
  method: 'POST',
});

const DELETE_BLUEPRINT = (blueprintId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}`,
  method: 'DELETE',
});

const CREATE_NEW_SERVICE =(blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services`,
  method: 'POST'
});

const UPDATE_SERVICE_POSITION= (blueprintId, serviceId, version) =>({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}`,
  method: 'PUT',
});

const registerUser = (name, otp, email, password) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'register',
    method: 'post',
    data: {
      name,
      otp,
      email,
      password,
    }
  };
};

const loginUser = (email, password) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'login',
    method: 'post',
    data: {
      email,
      password,
    }
  };
};

const validateUser = (token) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'validate',
    method: 'get',
    headers: {
      authorization: token
    }
  };

};

const generateOtp = (email) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'generate/otp',
    method: 'post',
    data: {
      email,
    }
  };
};

const GET_SYSTEM_COMPONENTS = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/plugins/types/SystemComponent',
  method: 'GET',
});

const GET_DEPLOYMENT_STACK = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/plugins/types/DeploymentStack',
  method: 'GET',
});

const GET_PLUGIN_BY_ID = (pluginId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/plugin/${pluginId}`,
  method: 'GET',
});


const CONNECT_SERVICES = (blueprintId,serviceId,version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/connections`,
  method: 'POST',

});

const DISCONNECT_SERVICES = (blueprintId,serviceId,version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/connections`,
  method: 'DELETE',
});

const GET_SERVICE_BY_ID = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}`,
  method: 'GET',
});

const UPDATE_CONFIGURATIONS = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/configurations`,
  method: 'PUT',
});

const UPDATE_METADATA = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/metadata`,
  method: 'PUT',
});

const DELETE_SERVICE_BY_ID = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}`,
  method: 'DELETE',
});

const UPDATE_BLUEPRINT_METADATA = (blueprintId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}`,
  method: 'PUT',
});

const SEARCH_CONFIGURATIONS = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/configurations`,
  method: 'GET',
});

const GET_ALL_CLUSTERS = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/clusters',
  method: 'GET',
});

const GET_CLUSTER_BY_ID = (clusterId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}`,
  method: 'GET',
});

const CREATE_CLUSTER = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/clusters',
  method: 'POST',
});

const GET_ALL_TASKS= () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/tasks',
  method: 'GET',
  
});

const GET_TASKS_BY_ID = (id) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/tasks/${id}`,
  method: 'GET',
});

const DOWNLOAD_BOILERPLATE = (objectName) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/minio/boilerplate/${objectName}`,
  method: 'GET',
});

const DELETE_CLUSTER = (clusterId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}`,
  method: 'DELETE',
});

const EXPORT_BLUEPRINT = (blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/exports`,
  method: 'POST',
});

const DEPLOY_CLUSTER = (clusterId,blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}/blueprints/${blueprintId}/versions/${version}/deploy`,
  method: 'POST',
});

const TEST_CONNECTION = (clusterId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}/testConnection`,
  method: 'POST',
});

export default { 
  generateOtp,
  registerUser,
  loginUser,
  validateUser,
  GET_ALL_BLUEPRINTS,
  GET_BLUEPRINT_METADATA,
  CREATE_BLUEPRINT,
  DELETE_BLUEPRINT,
  GET_SYSTEM_COMPONENTS,
  GET_SERVICES_BY_BLUEPRINT_ID,
  GET_SERVICE_BY_ID,
  UPDATE_CONFIGURATIONS,
  UPDATE_METADATA,
  CREATE_NEW_SERVICE,
  DELETE_SERVICE_BY_ID,
  CONNECT_SERVICES,
  DISCONNECT_SERVICES,
  UPDATE_BLUEPRINT_METADATA,
  UPDATE_SERVICE_POSITION,
  SEARCH_CONFIGURATIONS,
  GET_ALL_CLUSTERS,
  GET_CLUSTER_BY_ID,
  CREATE_CLUSTER,
  GET_DEPLOYMENT_STACK,
  GET_ALL_TASKS,
  GET_TASKS_BY_ID,
  DOWNLOAD_BOILERPLATE,
  GET_IMAGES,
  GET_PLUGIN_BY_ID,
  DELETE_CLUSTER,
  EXPORT_BLUEPRINT,
  DEPLOY_CLUSTER,
  TEST_CONNECTION
};
