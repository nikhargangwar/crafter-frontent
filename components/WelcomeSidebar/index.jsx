import React, { useEffect, useState } from 'react';
import { Close, Delete } from '@mui/icons-material';
import { Alert, Box, Button, Divider, Snackbar, Typography} from '@mui/material/';
import PropTypes from 'prop-types';
import { makeRequest } from '../../utils';
import { apiEndPoints } from '../../constants';
import { useNavigate } from 'react-router-dom';
import { convertCamelCaseToCapital } from '../../utils/ServiceHandler';
import { DeploymentStackModal, DeployModal } from '..';

function WelcomeSidebar(props) {
  const {setShowSidebar, selectedBlueprintId, selectedClusterId} = props;
  const [metadata, setMetadata] = useState({});
  const [openModal, setOpenModal] = useState(false);
  const [clusterName, setClusterName] = useState('');
  const [version, setVersion] = useState('');
  const navigate = useNavigate();
  const token = localStorage.getItem('jwt_token');
  const [snackbar, setSnackbar] = useState({
    status: false,
    message: '',
    type: ''
  });

  useEffect(() => {
    if (selectedBlueprintId){
      makeRequest.makeRequest(apiEndPoints.GET_BLUEPRINT_METADATA(selectedBlueprintId), navigate, token).then((response) => {
        setMetadata(response.metadata);
        setVersion(response.versions[0]);
      });
    }
  }, [selectedBlueprintId]);

  useEffect(() => {
    if (selectedClusterId){
      makeRequest.makeRequest(apiEndPoints.GET_CLUSTER_BY_ID(selectedClusterId), navigate, token).then((response) => {
        setMetadata(response.metadata);
        setClusterName(response.clusterName);
      });
    }
  }, [selectedClusterId]);

  const handleDelete = () => {
    if (selectedBlueprintId){
      makeRequest.makeRequest(apiEndPoints.DELETE_BLUEPRINT(selectedBlueprintId), navigate, token).then(() => {
        setShowSidebar(false);
      });
    }
    if (selectedClusterId){
      makeRequest.makeRequest(apiEndPoints.DELETE_CLUSTER(selectedClusterId), navigate, token).then(() => {
        setShowSidebar(false);
      });
    }
  };

  const handleTestConnection = () => {
    if (clusterName === 'local'){
      makeRequest.makeRequest(apiEndPoints.TEST_CONNECTION(selectedClusterId), null, token).then(() => {
        setSnackbar({
          status: true,
          message: 'Connection Successful',
          type: 'success'
        });
      }).catch(() => {
        setSnackbar({
          status: true,
          message: 'Connection Failed',
          type: 'error'
        });
      });
    } else {
      setOpenModal(true);
    }
  };
  
  return (
    <Box sx={{
      backgroundColor: 'background.secondary',
    }} className=' w-2/5 h-screen py-5 pl-10 pr-5 flex flex-col justify-between'>
      {openModal && selectedBlueprintId && <DeploymentStackModal openModal={openModal} setOpenModal={setOpenModal} isEdit={true} />}
      {openModal && selectedClusterId && <DeployModal openModal={openModal} setOpenModal={setOpenModal} selectedClusterId={selectedClusterId} cluster={clusterName}/>}
      <div>
        <div className='flex justify-between w-full mb-2'>
          <h1 className='text-2xl font-bold text-white font-[Poppins]'>{selectedBlueprintId ? 'BLUEPRINT INFORMATION' : 'CLUSTER INFORMATION'}</h1>
          <Close className='cursor-pointer text-white' onClick = {() => setShowSidebar(false)}/>
        </div>
        <Divider sx={{bgcolor: 'primary.main', height: 2}} />
        <Box className='flex justify-end my-5'>
          {/* <Button variant="contained" size='small' sx={{bgcolor:'primary.main',color:'white',fontWeight:'bold', fontSize:'16px'}} startIcon={<Edit />} onClick={() => setOpenModal(true)}>
            Edit
          </Button> */}
          <Button variant="contained" size='small' sx={{bgcolor:'primary.main',color:'white',fontWeight:'bold', fontSize:'16px', ':hover' : {bgcolor: 'red'}}} startIcon={<Delete />} onClick={handleDelete}>
            Delete
          </Button>
        </Box>
        {Object.keys(metadata).map((key) => (
          <Box key={key} className='flex mb-5 justify-between items-center gap-4'>
            <Typography variant='h6' color='primary.main' fontSize={18} fontWeight='800' className='mb-2 w-1/2'>{convertCamelCaseToCapital(key).toLowerCase()}</Typography>
            <Typography variant='h6' color='white' fontSize={18} fontWeight='800' className='mb-2 px-2 rounded-lg text-center w-1/2'backgroundColor='background.textBox'>{metadata[key]}</Typography>
          </Box>))
        }
      </div>
      {selectedBlueprintId && <Box className='bottom-0 mt-5 self-end'>
        <Button variant="contained" sx={{bgcolor:'primary.main',color:'white',fontWeight:'bold', fontSize:'16px'}} className='mt-10' onClick={() => navigate(`/dashboard/${selectedBlueprintId}/v/${version}`)}>
        Open Blueprint
        </Button> 
      </Box>}
      {selectedClusterId && <Box className='bottom-0 mt-5 self-end'>
        <Button variant="contained" sx={{bgcolor:'primary.main',color:'white',fontWeight:'bold', fontSize:'16px'}} className='mt-10' onClick={handleTestConnection}>
        Test Connection
        </Button> 
      </Box>}
      <Snackbar open={snackbar.status} autoHideDuration={6000} onClose={() => setSnackbar({ ...snackbar, status: false })}>
        <Alert onClose={() => setSnackbar({ ...snackbar, status: false })} severity={snackbar.type} sx={{ width: '100%' }}>
          {snackbar.message}
        </Alert>
      </Snackbar>
    </Box>
  );
}
export default WelcomeSidebar;

WelcomeSidebar.propTypes = {
  setShowSidebar: PropTypes.func.isRequired,
  selectedBlueprintId: PropTypes.string,
  selectedClusterId: PropTypes.string,
};