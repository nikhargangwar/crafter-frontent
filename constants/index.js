import apiEndPoints from './apiEndPoints';
import apiRoutes from './apiRoutes';
import entryTypes from './entryTypes';
import arrayOfWordsForLandingPage from './keywords';
import sliderImages from './sliderImages';

export { apiEndPoints, apiRoutes, entryTypes, arrayOfWordsForLandingPage, sliderImages };
