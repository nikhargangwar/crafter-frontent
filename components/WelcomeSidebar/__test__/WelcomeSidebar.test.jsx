import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import WelcomeSidebar from '..';
import { makeRequest } from '../../../utils';
import { apiEndPoints } from '../../../constants';

const mockedNavigate = jest.fn();
jest.spyOn(window.localStorage.__proto__, 'getItem').mockReturnValue('jwt_token');
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedNavigate,
}));

beforeEach(() => {
  jest.spyOn(makeRequest, 'makeRequest').mockResolvedValue({
    metadata: {
      name: 'test',
      description: 'test',
      version: 'test',
    },
    versions: ['test']
  });
});

describe('WelcomeSidebar component', () => {
  const setShowSidebar = jest.fn();
  const selectedBlueprintId = 1;
  const selectedClusterId = 1;

  it('should render correctly', () => {
    const { asFragment } = render(
      <Router>
        <WelcomeSidebar setShowSidebar={setShowSidebar} selectedBlueprintId={selectedBlueprintId}/>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display blueprint information', async () => {
    render(<Router>
      <WelcomeSidebar setShowSidebar={setShowSidebar} selectedBlueprintId={selectedBlueprintId}/>
    </Router>);
    const rows = await screen.findAllByText('test');
    expect(rows).toHaveLength(3);
  });

  it('should navigate to dashboard when Open Blueprint button is clicked', async () => {
    render(<Router>
      <WelcomeSidebar setShowSidebar={setShowSidebar} selectedBlueprintId={selectedBlueprintId}/>
    </Router>);
    const button = screen.getByRole('button', { name: 'Open Blueprint' });
    await fireEvent.click(button);
    expect(mockedNavigate).toHaveBeenCalledWith(`/dashboard/${selectedBlueprintId}/v/`);
  });

  it('should call setShowSidebar when Close button is clicked', async () => {
    render(<Router>
      <WelcomeSidebar setShowSidebar={setShowSidebar} selectedBlueprintId={selectedBlueprintId}/>
    </Router>);
    const closeButton = screen.getByTestId('CloseIcon');
    await fireEvent.click(closeButton);
    expect(setShowSidebar).toHaveBeenCalledWith(false);
  });

  it('should call make request and close sidebar when delete button is clicked for blueprints ', async () => {
    render(<Router>
      <WelcomeSidebar setShowSidebar={setShowSidebar} selectedBlueprintId={selectedBlueprintId}/>
    </Router>);
    const deleteButton = screen.getByTestId('DeleteIcon');
    await fireEvent.click(deleteButton);
    expect(makeRequest.makeRequest).toHaveBeenLastCalledWith(apiEndPoints.DELETE_BLUEPRINT(selectedBlueprintId), mockedNavigate, undefined);
    expect(setShowSidebar).toHaveBeenCalledWith(false);
  });

  it('should call make request and close sidebar when delete button is clicked for clusters ', async () => {
    render(<Router>
      <WelcomeSidebar setShowSidebar={setShowSidebar} selectedClusterId={selectedClusterId}/>
    </Router>);
    const deleteButton = screen.getByTestId('DeleteIcon');
    await fireEvent.click(deleteButton);
    expect(makeRequest.makeRequest).toHaveBeenLastCalledWith(apiEndPoints.DELETE_CLUSTER(selectedClusterId), mockedNavigate, undefined);
    expect(setShowSidebar).toHaveBeenCalledWith(false);
  });
});