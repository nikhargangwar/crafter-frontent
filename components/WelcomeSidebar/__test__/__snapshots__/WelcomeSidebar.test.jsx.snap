// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeSidebar component should render correctly 1`] = `
<DocumentFragment>
  <div
    class=" w-2/5 h-screen py-5 pl-10 pr-5 flex flex-col justify-between MuiBox-root css-12hyt15"
  >
    <div>
      <div
        class="flex justify-between w-full mb-2"
      >
        <h1
          class="text-2xl font-bold text-white font-[Poppins]"
        >
          BLUEPRINT INFORMATION
        </h1>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cursor-pointer text-white css-i4bv87-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-dw7mth-MuiDivider-root"
      />
      <div
        class="flex justify-end my-5 MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-a7pq15-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DeleteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          Delete
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="bottom-0 mt-5 self-end MuiBox-root css-0"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium mt-10 css-u6tkr3-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Open Blueprint
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
